src/Catalog/Persistence/AzureStorage.cs

```diff
@@ -125,8 +124,46 @@ private StorageListItem GetStorageListItem(IListBlobItem listBlobItem)
                     Trace.WriteLine(String.Format("Saved uncompressed blob {0} to container {1}", blob.Uri.ToString(), _directory.Container.Name));
                 }
             }
+            await TryTakeBlobSnapshotAsync(blob);
         }
 
+        /// <summary>
+        /// Take one snapshot only if there is not any snapshot for the specific blob
+        /// This will prevent the blob to be deleted by a not intended delete action
+        /// </summary>
+        /// <param name="blob"></param>
+        /// <returns></returns>
+        private async Task<bool> TryTakeBlobSnapshotAsync(CloudBlockBlob blob)
+        {
+            if (blob == null )
+            {
+                //no action
+                return false;
+            }
+            Stopwatch sw = new Stopwatch();
+            sw.Start();
```

Comment:
`Stopwatch` has a static [`StartNew()`](https://msdn.microsoft.com/en-us/library/system.diagnostics.stopwatch.startnew(v=vs.110).aspx) method that does what these two lines do. Would save you one line of code :)